sdks/python/apache_beam/runners/direct/transform_evaluator.py

```diff
@@ -500,23 +523,21 @@ def __init__(
         input_committed_bundle,
         side_inputs)
     self.test_stream = applied_ptransform.transform
+    self.event_index = 0
```

Comment:
Removed this. Also, the is_done is needed because of the small cleanup. There used to be a weird C++ iterator style on the TestStream object to get the event. So there needs to be a small state variable from the process_elements call to the finish_bundle call informing itself if there are any more elements to process.